damnit/backend/extraction_control.py

```diff
@@ -274,11 +347,11 @@ def reprocess(runs, proposal=None, match=(), mock=False, watch=False, direct=Fal
     for req in reqs[1:]:
         req.update_vars = False
 
-    for prop, run in props_runs:
-        req = ExtractionRequest(run, prop, RunData.ALL, match=match, mock=mock)
-        if direct:
+    if direct:
+        for req in reqs:
             submitter.execute_direct(req)
-        elif watch:
+    elif watch:
+        for req in reqs:
             submitter.execute_in_slurm(req)
-        else:
-            submitter.submit(req)
+    else:
+        submitter.submit_multi(reqs, limit_running=limit_running)
```

Comment:
Is the old `.submit()` used anywhere else, or can we delete it?

Reply:
Good question. It's still used inside `extract_data`, when the Solaris job finishes and launches a Maxwell job. We could change that to use `submit_multi` and always make an array job with 1 task - but it would mean the job names in `squeue` are less useful, because they don't include the run number.

(The thing where Solaris jobs launch Maxwell jobs also means we're not strictly limiting how many extraction jobs can run simultaneously. I don't think there's any easy way around that, but I figure that staggering when jobs start should still mitigate the thundering herd anyway.)

Reply:
Alrighty, let's leave it then.